### ISSUE TITLE
[ENH] Import more estimators from hmmlearn

### ISSUE THREAD

**Issue by miraep8** [alan-turing-institute/sktime#3363]
[hmmlearn](https://github.com/hmmlearn/hmmlearn) is a nifty python package with a lot of different sci-kit learn like estimators for using HMM algorithms. 

We now have a simple way to wrap and thus import new HMM estimators into `sktime`, and I thought that it could be a great task for anyone new to `sktime` or open source in general to get started with if you are looking for an easy issue to contribute. 

The estimators which still need to be wrapped are:
  - [ ] CategoricalHMM
  - [ ] MultinomialHMM
  - [ ] PoissonHMM

Here is a rough guide for what needs to be done in order to wrap an estimator from `hmmlearn`:
   - in the `sktime.annotations.hmm_learn` file add your new estimator to the `__all__` attribute list near the top of the file.
   - make a new class for your relevant `hmmlearn` estimator.
   - copy over the docstring from `hmmlearn` into your new class.  [Here](https://github.com/hmmlearn/hmmlearn/blob/main/lib/hmmlearn/hmm.py) is where the code lives on Github and you can check it out there.  (may need to look into both the `__init__` function and the main class docstring in order to get both the `Parameters` section and the `Attributes` slash general description. 
   - update the `__init__` and `_fit` functions to properly collect and store all the attributes needed for your wrapped estimator.  In `_fit` pass these parameters to a wrapped version of your estimator. 
   - update the docstring example
   - update the get_test_params method with some relevant params for your new estimator.
   - add a quick test in test_hmm_learn that gaurantees your new wrapped estimator agrees with the estimator it is wrapping in `hmmlearn`.  It will be important to make sure both implementations have the same random state. 
   
   To see an example of what this should look like please check out my recent PR (#3362) for GMMHMM which has all these steps in it.  If you would like top take on an estimator just comment in this issue so other people know you are working on it! :)

**Comment by kejsitake** [comment]
Hi! Since no one else has taken over, I am planning to work on importing these estimators this weekend, after a paper deadline tomorrow!

**Comment by miraep8** [comment]
Fantastic, thank you! :D  Let me know if you have any questions on this and I would be happy to review your PR when its ready, just ping me

**Comment by JonathanBechtel** [comment]
So I think myself and a few co-workers are also going to work on this issue.  Seeing that issue #3597 was never resolved, and there does not seem to be any additional HMM modules imported into the library since this issue, we're planning on seeing if we can get some of these taken care of.  

Please let us know if there are any recommendations about how best to proceed.

**Comment by miraep8** [comment]
I would recommend starting with the HMM modules other than `CategoricalHMM` at the moment (as @kejsitake still has a draft PR open on that at the moment). Please ping me when you get a PR together! :)

**Comment by JonathanBechtel** [comment]
Thank you!  We are working on the PoissonHMM right now, and if that goes well we'll do the MultinomialHMM too.  We're planning on attending the dev-standup tomorrow if we need help and get the PR taken care of ASAP.

**Comment by JonathanBechtel** [comment]
@miraep8 We are working on multinomial & poisson right now, but had a question about unit tests.  Right now the unit tests use a function called `piecewise_normal`, would it be sufficient to use this for unit tests for MultinomialHMM & PoissonHMM's too, or is it preferred we create the corresponding functions `piecewise_multinomial` and `piecewise_poisson` and use that for the datagen?  Thank you!

**Comment by miraep8** [comment]
I think the question with unit tests is always - what are you actually trying to test? :)  For example - are you interested in how these HMM's fit a piecewise normal data set, in that case then yes!  But I suspect you might be more interested in - how well do these HMM's fit the data they are trying to describe (which in this case would be a `piecewise_poisson` as you suggested!)

If going down that route - here would be my suggestion: 

- make a separate branch/PR for those new data gen functions you described!  Then once that is merged you can make the PRs for `PossionHMM` and `MultinomialHMM`.  Of course feel free to ping me when you make that PR.

Cheers!

**Comment by JonathanBechtel** [comment]
Thanks, appreciated.  Question:  I'm looking through the `tests` folder and don't see anything specifically for the `datagen` functions.  Is there a place where we should add a unit test for the functions mentioned above?

**Comment by Pyyim** [comment]
@miraep8 We have just submitted a PR for the piecewise_poisson as you suggested. #4078

Thank you.

**Comment by JonathanBechtel** [comment]
Also wrote another PR for `piecewise_multinomial`

**Comment by miraep8** [comment]
Nice work!  And I think you all actually suggested it not me! :)  Will check it out now, and thanks a lot for your help getting these estimators ported!

**Comment by miraep8** [comment]
Oh also - to answer your question - sure tests are always a good idea! :) We very well might not have them for the existing datagen functions, but that shouldn't stop you from adding some for your own!  I think the most logical place to put this would be sktime -> annotation -> tests and then make a new file called `datagen_tests.py`.  Thank you for spotting this!

**Comment by miraep8** [comment]
One other alternative to add a quick test (if you just want to see that the code can be imported and generate the results you expect) is to just use the docs tests. I see that you both skipped the docs tests in your PRs, which is also what `piecewise_normal` does, so not necessarily bad! (although I actually forget why we skip the docs test there, perhaps @lmmentel can remind me?).  To see an example that doesn't skip the docs tests can check out the `piecewise_normal_multivariate` instead.

**Comment by lmmentel** [comment]
Thanks for picking up the work @JonathanBechtel and @Pyyim! I don't recall exactly what was the issue with doctests but probably reproducing random samples and/or getting consistent predictions based on those. At this point if you add a test that checks if the dimensions of the produced samples are OK it will already be an improvement on top of what's there.
